### ISSUE TITLE
netty: out of memory error while uploading large files

### ISSUE THREAD

**Issue by jknack** [jooby-project/jooby#399]
File upload is 1.8GB and Netty failed with this stacktrace:

```
Caused by: java.lang.OutOfMemoryError: Java heap space
    at io.netty.buffer.UnpooledHeapByteBuf.copy(UnpooledHeapByteBuf.java:473) ~[netty-buffer-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.readFileUploadByteMultipart(HttpPostMultipartRequestDecoder.java:1492) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.getFileUpload(HttpPostMultipartRequestDecoder.java:902) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.decodeMultipart(HttpPostMultipartRequestDecoder.java:563) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.findMultipartDisposition(HttpPostMultipartRequestDecoder.java:804) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.decodeMultipart(HttpPostMultipartRequestDecoder.java:501) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.findMultipartDelimiter(HttpPostMultipartRequestDecoder.java:657) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.decodeMultipart(HttpPostMultipartRequestDecoder.java:488) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.parseBodyMultipart(HttpPostMultipartRequestDecoder.java:453) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.parseBody(HttpPostMultipartRequestDecoder.java:422) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.offer(HttpPostMultipartRequestDecoder.java:347) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.<init>(HttpPostMultipartRequestDecoder.java:193) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostRequestDecoder.<init>(HttpPostRequestDecoder.java:97) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
    at io.netty.handler.codec.http.multipart.HttpPostRequestDecoder.<init>(HttpPostRequestDecoder.java:68) ~[netty-codec-http-4.1.0.CR7.jar:4.1.0.CR7]
```

There is an open bug at netty too: https://github.com/netty/netty/issues/3559

If your app requires large file upload a better option for now is:
- http://jooby.org/doc/undertow/ or
- http://jooby.org/doc/jetty/

**Comment by elaatifi** [comment]
This is exactly why I tried undertow :)

**Comment by jknack** [comment]
sad :(... we deploy a new app and we found this.

**Comment by danechitoaie** [comment]
Could we maybe have Undertow as default server? Or was Netty better and that's what it was chosen as default?

**Comment by jknack** [comment]
Fixed in 2.0.0
